dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/hibernate/HibernateIdentifiableObjectStore.java

```diff
@@ -128,51 +128,45 @@ private void save( T object, User user, boolean clearSharing )
     {
         String username = user != null ? user.getUsername() : "system-process";
 
-        if ( IdentifiableObject.class.isAssignableFrom( HibernateProxyUtils.getRealClass( object ) ) )
```

Comment:
FYI: there were a a few similar checks but `T` is already bound to `BaseIdentifiableObject` so check must always be true.